planner/core/logical_plans.go

```diff
@@ -151,6 +151,12 @@ type LogicalJoin struct {
 	equalCondOutCnt float64
 }
 
+// Clone clones a LogicalJoin struct.
+func (p *LogicalJoin) Clone() *LogicalJoin {
```

Comment:
We can rename it to `Shallow` indicating that it's a shallow copy instead of deep clone.